File: java/Euler2.java
public class Euler2 {
    // https://projecteuler.net/problem=2
    
    public static void main(String[] args) {
        System.out.printf("The sum of the even-valued terms whose values do not exceed 4,000,000 is %d.", fibonacciSum());
    }

    public static int fibonacciSum() {
        int a = 0, b = 1, c = 2;
        int sum = 0;

        while (b < 4000000) {
            a = b;
            b = c;
            c = a + b;

            if (a % 2 == 0) {
                sum += a;
            }
        }

        return sum;
    }
}

File: python/euler2.py
# https://projecteuler.net/problem=2

limit = 4000000  # Do not consider Fibonacci sequence terms greater than this number
a, b, c = 0, 1, 2  # First three terms in Fibonacci sequence
sum = 0

while b <= limit:
    a = b  # a becomes b
    b = c  # b becomes c
    c = a + b  # c becomes the sum of a and b

    if a % 2 == 0:  # If a is even
        sum += a  # Add to sum

print(f'The sum of the even-valued terms whose values do not exceed {limit}, is {sum}.')